pkg/apis/externaldns/validation/validation.go

```diff
@@ -86,6 +86,16 @@ func ValidateConfig(cfg *externaldns.Config) error {
 		if cfg.RFC2136MinTTL < 0 {
 			return errors.New("TTL specified for rfc2136 is negative")
 		}
+
+		if cfg.RFC2136Insecure && cfg.RFC2136GSSTSIG {
+			return errors.New("--rfc2136-insecure and --rfc2136-gss-tsig are mutually exclusive arguments")
+		}
+
+		if cfg.RFC2136GSSTSIG {
+			if cfg.RFC2136KerberosPassword == "" || cfg.RFC2136KerberosUsername == "" {
+				return errors.New("--rfc2136-kerberos-username and --rfc2136-kerberos-password both required when specifying --rfc2136-gss-tsig option")
```

Comment:
Not ideal to take a password via a CLI parameter, as running "ps -Af" or similar on the host will reveal the password. Probably would want to take it in as a file parameter and read it.

An alternative would be to take a keytab path as input and directly use that. Would probably be a more "productionised" way to deploy this.

Reply:
@randomvariable 100% agree with you.  However, I made this decision for 2 reasons:

1) A good portion of the other providers are already using username/password to authenticate with their providers.  Granted, username/password for an AD domain environment is probably more privileged (if not configured correctly) than simply a DNS provider username/password.  Using username/password, although not ideal for many environments, is a very simple way to get something working.  I did this with the understanding that this codebase would likely migrate to something more secure at a later time.  My goal for this simply just MVP.  See https://github.com/kubernetes-sigs/external-dns/blob/master/pkg/apis/externaldns/types.go for examples of providers using username/password .

2) The underlying library package which negotiates the security context and grabs the keydata for the DNS message exchange for GSS-TSIG does not support keytab.  You can see this at https://github.com/bodgit/tsig/blob/master/gss/sspi.go#L200-L203 .  I actually started down the road of keytab to begin with, but had to back off, as I was focused on POC/MVP first, with the understanding that we can improve upon this later, perhaps even a PR into the upstream library.  You can see my thought process when I originally ran across this at https://github.com/scottd018/external-dns-rfc3645/blob/master/provider/rfc3645.go#L91-L96.

Hope that helps outline the decision making process.  Thanks for the input!  :)

Reply:
Noted, thanks!